Improvements to the mock cloud database

mbrandonw opened the pull request:
There was a bit of behavior not yet captured by our mock cloud database that this PR plugs. If you save a record in the shared database that is not apart of a hierarchy that leads to a shared root record, iCloud throws a `CKError.permissionFailure`. We now replicate that behavior.

mbrandonw committed:
Private tables should always be saved in private db.

mbrandonw committed:
snapshots

mbrandonw committed:
Handle shared database more correctly.

mbrandonw committed:
wip

mbrandonw committed:
fix test

mbrandonw committed:
Merge remote-tracking branch 'origin/main' into improved-mock-cloud-database

mbrandonw committed:
add another test

stephencelis commented on Tests/SQLiteDataTests/CloudKitTests/MockCloudDatabaseTests.swift:
```diff
@@ -421,6 +423,95 @@
+
+        print(saveResults)
+
+        assertInlineSnapshot(of: container, as: .customDump) {
```
```suggestion
        )

        assertInlineSnapshot(of: container, as: .customDump) {
```

stephencelis committed:
Update Tests/SQLiteDataTests/CloudKitTests/MockCloudDatabaseTests.swift

stephencelis committed:
Merge branch 'main' into improved-mock-cloud-database

RobbieClarken committed:
Fix id column type in documentation example schema to prevent error (#322)

When the `id` column is defined as `INT NOT NULL PRIMARY KEY AUTOINCREMENT`
sqlite will throw an error:

> AUTOINCREMENT is only allowed on an INTEGER PRIMARY KEY

mbrandonw committed:
fixes

mbrandonw committed:
format

mbrandonw committed:
Merge remote-tracking branch 'origin/main' into improved-mock-cloud-database
